signal/report.go

```diff
@@ -0,0 +1,62 @@
+package signal
+
+import (
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/dcos/dcos-signal/config"
+)
+
+type Report interface {
```

Comment:
why is this an interface vs. a struct (that you can embed where needed) with public fields? not used to seeing a list of Get/Set funcs like this